### PR TITLE
Some fixups for tpm-eventlog tool

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/tpm-eventlog/fu-tpm-eventlog.c [hunk 1]
```diff
@@ -50,19 +51,29 @@ fu_tmp_eventlog_process (const gchar *fn, gint pcr, GError **error)
...
+	for (guint8 i = 0; i <= max_pcr; i++) {
 		g_autoptr(GPtrArray) pcrs = fu_tpm_eventlog_calc_checksums (items, i, NULL);
 		if (pcrs == NULL)
 			continue;
 		for (guint j = 0; j < pcrs->len; j++) {
-			const gchar *csum = g_ptr_array_index (pcrs, j);
-			g_autofree gchar *title = g_strdup_printf ("%x", i);
+			const gchar *csum;
+			g_autofree gchar *title = NULL;
+			if (pcr >=0 && i != (guint) pcr)
```
Comment (hughsie) [inline plugins/tpm-eventlog/fu-tpm-eventlog.c:73]: space before the 0 please

File: plugins/tpm-eventlog/fu-tpm-eventlog.c [hunk 2]
```diff
@@ -50,19 +51,29 @@ fu_tmp_eventlog_process (const gchar *fn, gint pcr, GError **error)
...
 		FuTpmEventlogItem *item = g_ptr_array_index (items, i);
+		if (item->pcr > max_pcr)
+			max_pcr = item->pcr;
 		if (pcr >= 0 && item->pcr != pcr)
 			continue;
 		fu_tpm_eventlog_item_to_string (item, 0, str);
 		g_string_append (str, "\n");
 	}
+	if (pcr > max_pcr) {
+		g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA, "invalid PCR %d", pcr);
```
Comment (hughsie) [inline plugins/tpm-eventlog/fu-tpm-eventlog.c:62]: line is a bit long

Comment (hughsie) [reply plugins/tpm-eventlog/fu-tpm-eventlog.c:62]: Also, PCR should be %x, no?

File: plugins/tpm-eventlog/fu-tpm-eventlog.c (hunk 1 above)
Comment (hughsie) [reply plugins/tpm-eventlog/fu-tpm-eventlog.c:73]: i think you can do `const gchar *csum = g_ptr_array_index (pcrs, j);` still here, it seems a bit of a microoptimisation to move it down :)

File: plugins/tpm-eventlog/fu-tpm-eventlog.c (hunk 2 above)
Comment (superm1) [reply plugins/tpm-eventlog/fu-tpm-eventlog.c:62]: Is it? I guess right now we're only going up to PCR 9 on my testing system, but I've never seen a PCR above referred to as 0xa.

Comment (hughsie) [reply plugins/tpm-eventlog/fu-tpm-eventlog.c:62]: ohh true! ignore me, sorry.

**Pull request merged**